src/lib.rs

```diff
@@ -510,6 +518,27 @@ pub fn is_loadbalancer_enabled(input: &str) -> bool {
     input.trim().eq_ignore_ascii_case("true")
 }
 
+/// Parse the syslog endpoint from metadata
+///
+/// Accepts hostname or IP address with port. Returns None if the input is empty.
+///
+/// # Arguments
+///
+/// * `input` - The syslog endpoint string (e.g., "syslog.example.com:514" or "10.11.28.101:30514")
+///
+/// # Returns
+///
+/// The endpoint as a String, or None if empty
+pub fn parse_syslog_endpoint(input: &str) -> Option<String> {
+    let trimmed = input.trim();
+    if trimmed.is_empty() {
+        return None;
+    }
+
+    // Just return the trimmed input - we trust the user's input
+    Some(trimmed.to_string())
```

Comment:
So this returned string goes *where* exactly?  Something eventually has to call `getipnodebyname`()...